filebeat/harvester/processor/line.go

```diff
@@ -0,0 +1,90 @@
+package processor
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// LineSource produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type LineSource struct {
+	reader *encoding.LineReader
+}
```

Comment:
For getting proper multiline json support one has to implement `JsonProducer` adapter wrapping io.Reader with json.Decoder. dealing with encoding might still be required, as golang requires UTF-8, but json can be UTF-16 as well.